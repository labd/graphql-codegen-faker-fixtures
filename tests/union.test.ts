import { plugin } from "../dist";
import { buildSchema } from "graphql/utilities";
import { parse } from "graphql";

const schema = /* GraphQL */ `
  type Cat {
    name: String
    color: String
  }

  type Dog {
    name: String
    isGoodBoy: Boolean
  }

  union Pet = Cat | Dog
`;

const document = /* GraphQL */ `
  fragment cat on Cat {
    name
    color
  }

  fragment dog on Dog {
    name
    isGoodBoy
  }

  fragment pet on Pet {
    ...cat
    ...dog
  }
`;

describe("union", () => {
  it("should create a builder for a fragment that uses a union type", () => {
    const response = plugin(
      buildSchema(schema),
      [{ document: parse(document) }],
      { buildersOnly: true },
    );
    expect(response).toMatchSnapshot();

    expect(response).toContain(
      `faker.helpers.arrayElement([
          fakeCat(),fakeDog()
        ])`,
    );
  });
});
