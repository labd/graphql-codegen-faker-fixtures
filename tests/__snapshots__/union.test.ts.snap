// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`union should create a builder for a fragment that uses a union type 1`] = `
"
    export const fakeCat = (override?:DeepPartial<types.CatFragment>):types.CatFragment => {
      const fixture:types.CatFragment = {
        name: faker.lorem.words(),color: faker.lorem.words()
      }

      return override
        ? merge(fixture, override)
        : fixture
    }
  

    export const fakeDog = (override?:DeepPartial<types.DogFragment>):types.DogFragment => {
      const fixture:types.DogFragment = {
        name: faker.lorem.words(),isGoodBoy: faker.datatype.boolean()
      }

      return override
        ? merge(fixture, override)
        : fixture
    }
  

      export const fakePet = ():types.PetFragment =>
        faker.helpers.arrayElement([
          fakeCat(),fakeDog()
        ])
    "
`;
